### ISSUE TITLE
Support auto-suggestions for searching

### ISSUE THREAD

**Issue by LegoStormtroopr** [aristotle-mdr/aristotle-metadata-registry#700]

**Comment by DeKaN** [comment]
Can you add some info? I can try to do this :)

**Comment by LegoStormtroopr** [comment]
Yup, so the basic gist is - on the search page (you can see an example here - https://registry.aristotlemetadata.com/search), it would be good to add some autocomplete functionality. Django-Haystack has an 'autocomplete' API that should make this easy and search index independent.

In the search box, it would be good if the current text was used to give a list of autocomplete options from that API. This should involve a little javascript as well as some python, and that can probably be added to [`aristotle_mdr/static/aristotle_mdr/aristotle.search.js`](https://github.com/aristotle-mdr/aristotle-metadata-registry/blob/develop/aristotle_mdr/static/aristotle_mdr/aristotle.search.js)

**Comment by derrix060** [comment]
You can do this on the client-side,there is a good js framework [Typeahead](http://twitter.github.io/typeahead.js/) to do this, and you can see one example [here](http://twitter.github.io/typeahead.js/examples/). There is a data structure to help this auto search, the name is *Trie*, and you can search more about it. I hope that it helped you.

**Comment by LegoStormtroopr** [comment]
I've looked at Typeahead before and it seems to be abandoned - it looks like there are some active forks, but I'm still wary about adding a new dependency.

Aristotle already has django-autocomplete-light and haystack, so I think those should be enough to bolt them together. Might need a `/search/autocomplete` endpoint that does some lookups. There might need to be an Ngam field added to the search index

I don't have a ton of time to work on this, but it's a requested feature, so @derrix060 if you are happy to lead a proof-of-concept with typeahead, we can look at integrating the search engine underneath later. Maybe look at https://github.com/corejavascript/typeahead.js since it has most recent updates and such.

**Comment by DeKaN** [comment]
Alternative: http://django-haystack.readthedocs.io/en/master/autocomplete.html
